packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -86,13 +92,30 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
   /// Scrollbar will be rectangular if [radius] is null.
   final Radius radius;
 
-  /// The smallest size the scrollbar can shrink to when the total scrollable
-  /// extent is large and the current visible viewport is small, and the
-  /// viewport is not overscrolled. Mustn't be null.
+  /// The amount of space by which to inset the scrollbar's start and end, as
+  /// well as its side to the nearest edge, in pixels.
+  ///
+  /// This is typically set to the current [MediaQueryData.padding] to avoid
+  /// partial obstructions such as display notches. If you want additonal
+  /// margins around the scrollbar, see [mainAxisMargin] or [crossAxisMargin].
+  ///
+  /// Defaults to [EdgeInsets.zero]. Must not be null and offsets from all four
+  /// directions must be greater than or equal to zero.
+  final EdgeInsets padding;
+
+  /// The preferred smallest size the scrollbar can shrink to when the total
+  /// scrollable extent is large and the current visible viewport is small, and the
+  /// viewport is not overscrolled.
+  ///
+  /// Mustn't be null and the value is typically greater than or equal to
+  /// [minOverscrollLength].
```

Comment:
In current implementation when `minOverscrollLength > minLength`, the minimum normal scrolling length is in fact `minOverscrollLength`: 
```
    double thumbExtent = math.min(effectiveViewport, minOverscrollLength);
    if (before > 0 && after > 0) {
      thumbExtent = math.max(	      thumbExtent = math.max(
        minLength,
        thumbExtent,	        thumbExtent,
        viewport * fractionVisible - 2 * mainAxisMargin,	
      );	      );
      // Thumb extent is no smaller than minLength if scrolling normally.	    
}
```
Should we enforce `minOverscrollLength <= minLength`?

Reply:
Also the value is *preferred* because it's only a hint, i.e. if it's set to a unreasonably large value, or when `mainAxisMargin` gets too large we won't be able to stick to these min values

Reply:
Ya, I think it's ok to generally add more enforcements to make sure that the relationships between minLength, minOverscrollLength are respected and describe edge cases.